pkg/netscan/util_test.go

```diff
@@ -0,0 +1,142 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2022 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package netscan
+
+import (
+	"context"
+	"encoding/binary"
+	"fmt"
+	"net"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"testing"
+)
+
+type inetTest struct {
+	inet  string
+	first string
+	last  string
+	size  uint32
+	err   bool
+}
+
+func mockIpWorker(ipCh <-chan uint32, result *inetTest) {
+	ip := net.IP([]byte{0, 0, 0, 0})
+	var last uint32
+
+	for a := range ipCh {
+		atomic.AddUint32(&result.size, 1)
+		atomic.StoreUint32(&last, a)
+
+		if result.first == "" {
+			binary.BigEndian.PutUint32(ip, a)
+			result.first = ip.String()
+		}
+		binary.BigEndian.PutUint32(ip, last)
+		result.last = ip.String()
+	}
+
+}
+
+func ipGeneratorTest(input inetTest) (result inetTest) {
+	var wg sync.WaitGroup
+	ipCh := make(chan uint32, input.size)
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		mockIpWorker(ipCh, &result)
+	}()
+
+	_, inet, err := net.ParseCIDR(input.inet)
+	if err != nil {
+		result.err = true
+		return result
+	}
+
+	ipGenerator(context.Background(), inet, ipCh)
+	close(ipCh)
+	wg.Wait()
+
+	return result
+}
+
+// TestIpGenerator calls the ip generator and validates that the first ip, last ip, and size
+// match the expected values.
+func TestIpGenerator(t *testing.T) {
+	tests := []inetTest{
```

Comment:
Need an error test case. Looks like test code looks for it, but missing actual tests case.